### ISSUE TITLE
SocketsHttpHandler doesn't support receiving multiple "100 continue" status responses

### ISSUE THREAD

**Issue by mconnew** [dotnet/corefx#31413]
According to [RFC7231 section 6.2](https://tools.ietf.org/html/rfc7231#section-6.2), multiple "100 continue" status responses should be supported:

> A client MUST be able to parse one or more 1xx responses received
> prior to a final response, even if the client does not expect one.  A
> user agent MAY ignore unexpected 1xx responses.

WCF has a customer who is communicating with a SOAP service which is returning multiple 100 status responses. The second 100 status response is being returned to WCF as the final response. Additionally, after looking at the code, I don't believe SocketsHttpHandler supports receiving a 100 status response when the `Expect: 100-continue` header was absent in the outgoing request. This could be as a result of an intermediate proxy having added an Expect header when forwarding the request and not removing the 100 status response on the reply. The WCF issue tracking this is dotnet/wcf#3048

**Comment by geoffkizer** [comment]
I think we should do this for 1xx status codes generally (not just 100 Continue).

Per RFC 7231:

>   A client MUST be able to parse one or more 1xx responses received
   prior to a final response, even if the client does not expect one.  A
   user agent MAY ignore unexpected 1xx responses.

We should just parse and discard any 1xx responses, including 100 Continue when not expected.

**Comment by caesar1995** [comment]
> I don't believe SocketsHttpHandler supports receiving a 100 status response when the Expect: 100-continue header was absent in the outgoing request.

RFC specifies that in this case, client can discard interim 100 continue response. Did I misunderstand something?

Per RFC 7231 section 6.2.1.  100 Continue:

```
If the request did not contain an Expect header field containing the
   100-continue expectation, the client can simply discard this interim
   response.
```

**Comment by caesar1995** [comment]
Fixed in #34507.
